firefox-ios/Client/Assets/CC_Script/AutofillTelemetry.sys.mjs

```diff
@@ -32,33 +32,15 @@ class AutofillTelemetryBase {
     extra[this.SUPPORTED_FIELDS[key]] = value;
   }
 
-  /**
-   * Building the extra keys object that is included in the Legacy Telemetry event `cc_form_v2`
-   * or `address_form` event and the Glean event `cc_form`, and `address_form`.
-   * It indicates the detected credit card or address fields and which method (autocomplete property, regular expression heuristics or fathom) identified them.
-   *
-   * @param {Array<object>} fieldDetails fieldDetails to extract which fields were identified and how
-   * @param {string} undetected Default value when a field is not detected: 'undetected' (Glean) and 'false' in (Legacy)
-   * @param {string} autocomplete Value when a field is identified with autocomplete property: 'autocomplete' (Glean), 'true' (Legacy)
-   * @param {string} regexp Value when a field is identified with regex expression heuristics: 'regexp' (Glean), '0' (Legacy)
-   * @param {boolean} includeMultiPart Include multi part data or not
-   * @returns {object} Extra keys to include in the form event
-   */
-  #buildFormDetectedEventExtra(
-    fieldDetails,
-    undetected,
-    autocomplete,
-    regexp,
-    includeMultiPart
-  ) {
```

Comment:
@issammani looks like decent amount of changes here, could you double check before merging please 🙏🏼